### PR TITLE
docs(dialogflow): Add section on streaming methods to the migration guide

### PR DESCRIPTION
**Pull request by dazuma**
Dialogflow has streaming methods, and I neglected to add this section to its migration guide.

### CODE REVIEW

File: google-cloud-dialogflow/MIGRATING.md [hunk 1]
```diff
@@ -285,6 +290,94 @@ def foo
...
+end
+```
+
+New:
+```
+client = Google::Cloud::Dialogflow.sessions
+
+# Create a request stream, initiate the call, and get a response stream.
+request_stream = Gapic::StreamInput.new
+response_stream = client.streaming_detect_response request_stream
```
Comment (quartzmo) [inline google-cloud-dialogflow/MIGRATING.md:343]: Shouldn't this be `client. streaming_detect_intent`?

Comment (dazuma) [reply google-cloud-dialogflow/MIGRATING.md:343]: Thanks for catching that. Fixed.

**Comment by dazuma** [comment]
@quartzmo Regarding the streaming example, that's a good question and I don't have a good answer at the moment. The samples do include a streaming example that follows this pattern (see https://github.com/googleapis/google-cloud-ruby/pull/5813/files#diff-e9fd630a37b10fbeef3e82d16080cdc0). But generating rich samples in the docs and unit tests in the microgenerator is still a challenging open question.

**Comment by quartzmo** [comment]
> The samples do include a streaming example that follows this pattern (see https://github.com/googleapis/google-cloud-ruby/pull/5813/files#diff-e9fd630a37b10fbeef3e82d16080cdc0). 

As a rule, all of the examples in the `OVERVIEW.md` guides are manually copied without modification from in-code samples that are tested with doctest. It seems like whenever possible, copying directly, without modification from the tested samples would also be a good approach with the migration guides.

**Pull request merged**